src/main/kotlin/me/serce/solidity/lang/psi/impl/mixins.kt

```
@@ -270,28 +270,31 @@ abstract class SolStructDefMixin : SolStubbedNamedElementImpl<SolStructDefStub>,
 }
 
 abstract class SolFunctionCallMixin(node: ASTNode) : SolNamedElementImpl(node), SolFunctionCallElement, SolFunctionCallExpression {
-  override fun getBaseAndReferenceNameElement(): Pair<SolExpression?, PsiElement> {
-    return when (val expr = expression) {
+
+  private fun getReferenceNameElement(expr: SolExpression): PsiElement {
+    return when (expr) {
       is SolPrimaryExpression ->
-        expr.varLiteral?.let { Pair(null, it) }
-          ?: expr.elementaryTypeName?.let { Pair(null, it) }!!
+        expr.varLiteral ?: expr.elementaryTypeName!!
       is SolMemberAccessExpression ->
-        Pair(expr.expression, expr.identifier!!)
+        expr.identifier!!
       is SolFunctionCallExpression ->
-        Pair(expr.expression, expr.firstChild)
+        expr.firstChild
       is SolIndexAccessExpression ->
-        Pair(expr.expressionList.first(), expr.firstChild)
+        expr.firstChild
       is SolNewExpression ->
-        Pair(expr, expr.typeName as PsiElement)
-      else -> throw IllegalStateException("unable to extract reference name element from $this")
```

Doesn't seem to be very practical to handle all of the cases, and fail hard.